### PR TITLE
Add `Debug` Code Lens

### PR DESCRIPTION
**Pull request by anderseknert**
Following the excellent work on debugger support by @johanfylling, this PR introduces a code lens for debugging, which triggers a command that simply returns a launch configuration where the package or rule clicked is set as the entrypoint.

The code lens provider is now also rewritten in Rego.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     "OPA_EVAL_CAPABILITIES": "${workspacePath}/build/capabilities.json"
   },
   "opa.roots": [
+    "${workspaceFolder}/lab",
```
Comment (srenatus) [inline .vscode/settings.json:7]: Is this related to the PR?

**Review by srenatus — APPROVED** [review]
looks good to me

File: internal/lsp/server.go [hunk 2]
```diff
@@ -499,6 +499,52 @@ func (l *LanguageServer) StartCommandWorker(ctx context.Context) { // nolint:mai
...
 				// handle this ourselves as it's a rename and not a content edit
 				fixed = false
+			case "regal.debug":
+				if l.clientIdentifier != clients.IdentifierVSCode {
+					l.logError(errors.New("regal.debug is only supported in VSCode"))
+
+					break
+				}
+
+				if len(params.Arguments) != 3 {
```
Comment (johanfylling) [inline internal/lsp/server.go:509]: I don't see us reading arg `2` anywhere. Looks like it's the line number for `regal.eval`, which I suppose we don't need for debug. Do we still need to assert its presence?

File: bundle/regal/lsp/codelens/codelens.rego [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+_eval_lenses contains _rule_lens(rule, "regal.eval", "Evaluate") if {
+	some rule in ast.rules
+}
+
+_debug_lenses contains {
+	"range": location.to_range(result.ranged_location_from_text(input["package"]).location),
+	"command": {
+		"title": "Debug",
```
Comment (johanfylling) [inline bundle/regal/lsp/codelens/codelens.rego:43]: Does vscode support emojis? If it does, we could prefix this with a '🪲' 😄. 
There's probably plenty of setups where that would break, though.

File: bundle/regal/lsp/codelens/codelens.rego [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+			ast.ref_to_string(input["package"].path),
+			util.to_location_object(input["package"].location).row,
+		],
+	},
+}
+
+_debug_lenses contains _rule_lens(rule, "regal.debug", "Debug") if {
+	some rule in ast.rules
+
+	# no need to add a debug lens for a rule like `pi := 3.14``
```
Comment (johanfylling) [inline bundle/regal/lsp/codelens/codelens.rego:56]: 👍

File: bundle/regal/lsp/codelens/codelens.rego [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+_rule_lens(rule, command, title) := {
+	"range": location.to_range(result.ranged_location_from_text(rule).location),
+	"command": {
+		"title": title,
+		"command": command,
+		"arguments": [
+			input.regal.file.name, # regal ignore:external-reference
+			sprintf("%s.%s", [ast.ref_to_string(input["package"].path), ast.ref_static_to_string(rule.head.ref)]),
```
Comment (johanfylling) [inline bundle/regal/lsp/codelens/codelens.rego:67]: Does this strip the dynamic tail off of the rule's ref?

I suppose that it doesn't hurt that vars are transferred to the query, as I think the only side effect is that they will be unified for the result(?). Worst case, the result set will just be large for a large set of enumerated values.
I can even imagine this being a benefit, as it'd possibly include fewer secondary rules with intersecting paths, as these won't satisfy the vars 🤔 .

**Review by johanfylling — APPROVED** [review]
LGTM 👍 😃

File: bundle/regal/lsp/codelens/codelens.rego (hunk 3 above)
Comment (anderseknert) [reply bundle/regal/lsp/codelens/codelens.rego:43]: It does support that, and since we're only targeting VS Code for now, we could do it. Not sure this is intuitive though. Most people will probably think they got some dirt on their monitor 🤣

bugs

File: bundle/regal/lsp/codelens/codelens.rego (hunk 5 above)
Comment (anderseknert) [reply bundle/regal/lsp/codelens/codelens.rego:67]: Yes, `ast.ref_static_to_string` stops at the first dynamic part of a ref. Perhaps we can experiment with dynamic refs later, but I think this is the option that makes the most sense to me.

File: internal/lsp/server.go (hunk 2 above)
Comment (anderseknert) [reply internal/lsp/server.go:509]: No, you're right, we don't need to assert this here, but we're using the same outputs for both code lenses... and we'll want to check this to avoid getting out of bounds errors in case there aren't at least 2 args.

File: .vscode/settings.json (hunk 1 above)
Comment (anderseknert) [reply .vscode/settings.json:7]: No, thanks! Removed. Although it _is_ rather tricky to get it right with the "checkOnSave" feature when you want it outside of the main bundle but not the whole workspace... I'll guess I'll have to do some git magic to make sure this line remains unseen. I've removed it from the change now anyway.

File: bundle/regal/lsp/codelens/codelens.rego (hunk 3 above)
Comment (johanfylling) [reply bundle/regal/lsp/codelens/codelens.rego:43]: I was thinking something like `► Evaluate`|`🪲Debug`
